### DISCUSSION TITLE
✅ Add ability to hide data if widget is empty

### DISCUSSION THREAD

**Discussion by vladninja** [elementor/elementor#21423] [Feature Request]
### Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro )

### What problem is your feature request going to solve? Please describe.

When I am adding e.g. some Heading (static) + Loop Grid/Text/post Info widget with dynamic tag and save as a template in some pages when these widgets will be used may be the situation, when the heading will be present on the page, but some widgets with dynamic tags - no. 
So the only heading will be shown, which is not good.
Sometimes I have multiple sections like that on the page, so the user only see 6 headings with no info, it's confused.

### Describe the solution you'd like

If we have a new Container widget we probably can easily add some settings which can solve that problem.

My current solution is:
Create a container, place any static content + any dynamic widget (Gallery/Loop/Text/Post Info with dynamic tags etc), add some CSS class for that container which describe which type of the dynamic block inside. On the FE currently there is no any info about if some dynamic block was used. If the block is empty, Elementor just not render that on the FE and don't give any info about that, so I don't have any info about removing block from the DOM.
Maybe Elementor can provide that information in a some way or implement some settings to container which add ability to hide a container if there is no dynamic content.

### Describe alternatives you've considered

Add PHP (or JS) hooks for the containers like "Render Widget Content", but for the Containers, so the developers may control that manually via PHP before the render.

**Comment by nicholaszein** [comment]
Hello!

### 📢 We're happy to announce that the issue you raised was resolved in Elementor Pro v3.19.0, and improved with Elementor Pro 3.20.0! 🥳 

You can now use Display Conditions for widgets based on a variety of conditions, including date, time, user role, logged in/out, and others (more to come too). Check the [documentation for the new Display Conditions](https://elementor.com/help/show-and-hide-elements-on-a-page-display-conditions/) for widgets (Pro).

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
